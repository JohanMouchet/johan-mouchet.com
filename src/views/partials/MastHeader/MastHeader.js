// @flow

import * as React from "react";

const ___ = () => {
  return (
	<header className="p-mast-header">
		<div className="p-mast-header__inner">
			<h1 className="p-mast-header__heading">
				<span className="p-mast-header__half p-mast-header__half--left">
					<span className="p-mast-header__text-wrapper">
						<span className="p-mast-header__text">
							Johan
						</span>
					</span>
				</span>

				<span className="p-mast-header__line"></span>

				<span className="p-mast-header__half p-mast-header__half--right">
					<span className="p-mast-header__text-wrapper">
						<span className="p-mast-header__text">
							Mouchet
						</span>
					</span>
				</span>
			</h1>

			<h2 className="p-mast-header__tagline">Never stop exploring,<br className="p-mast-header__tagline-breaker"> never stop creating</h2>
		</div>

		{/* <div className="p-mast-header__language">
			<a className="p-mast-header__language-link" href="en/" hreflang="en">En</a> | <a className="p-mast-header__language-link" href="fr/" hreflang="fr">Fr</a>
		</div> */}
	</header>
	);
};

export default ___;
