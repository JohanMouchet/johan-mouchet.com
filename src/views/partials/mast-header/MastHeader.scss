@import "assets/styles/settings/_settings";
@import "assets/styles/tools/_tools";

.p-mast-header {
  $root: &;
  $skew-angle: 5deg;

  min-height: 400px;
  width: 100%;
  background: url("#{t-path(theme)}/header.jpg") left center no-repeat
    $s-color-primary;
  background-size: cover;
  text-shadow: 1px 1px 1px rgba($s-color-black, 0.3);
  overflow: hidden;
  position: relative;

  &::before {
    content: "";
    position: absolute;
    top: 0;
    left: 0;
    height: 100%;
    width: 100%;
    background-image: linear-gradient(
      transparent,
      rgba($s-color-primary-darker, 0.9) 90%
    );
  }

  & > * {
    position: relative;
  }

  &__inner {
    min-height: inherit;
    display: flex;
    flex-wrap: wrap;
    align-content: center;
  }

  /* Heading */
  &__heading {
    display: flex;
    align-items: center;
    width: 100%;
    margin-bottom: 0;
    color: $s-color-white;
    font-family: $s-font-family-tertiary;
    font-size: t-modular-scale(4);
    font-weight: 300;
    text-transform: uppercase;
    transition: font-size 0.5s;
    transform: skew(-$skew-angle);

    @media (min-width: t-breakpoint(sm)) {
      font-size: t-modular-scale(6);
    }

    @media (min-width: t-breakpoint(md)) {
      font-size: t-modular-scale(9);
    }
  }

  /* Line */
  &__line {
    height: t-vertical-rhythm(8, "rem");
    width: 2px;
    margin-left: 3px;
    margin-right: 3px;
    position: relative;

    &::after {
      content: "";
      position: absolute;
      top: 0;
      left: 0;
      height: 100%;
      width: 100%;
      background: currentColor;
      animation: line-animation 0.5s ease-out both;
      /* stylelint-disable max-nesting-depth */
      @keyframes line-animation {
        from {
          transform: scaleY(0);
        }

        to {
          transform: scaleY(100%);
        }
      }
      /* stylelint-enable max-nesting-depth */
    }
  }

  /* Halfs */
  &__half {
    flex-basis: 50%;
    max-width: 50%;
    overflow: hidden;
  }

  &__half--left {
    margin-bottom: t-vertical-rhythm(2, "rem");
    padding-right: 0.2em;
    text-align: right;

    #{$root}__text {
      animation: left-slide-animation 0.5s 0.5s ease-out both;
      /* stylelint-disable max-nesting-depth */
      @keyframes left-slide-animation {
        0% {
          transform: translateX(100%);
          opacity: 0;
        }

        50% {
          opacity: 1;
        }

        100% {
          transform: translateX(0);
        }
      }
      /* stylelint-enable max-nesting-depth */
    }
  }

  &__half--right {
    margin-top: t-vertical-rhythm(2, "rem");
    padding-left: 0.2em;

    #{$root}__text {
      animation: right-slide-animation 0.5s 0.5s ease-out both;
      /* stylelint-disable max-nesting-depth */
      @keyframes right-slide-animation {
        0% {
          transform: translateX(-100%);
          opacity: 0;
        }

        50% {
          opacity: 1;
        }

        100% {
          transform: translateX(0);
        }
      }
      /* stylelint-enable max-nesting-depth */
    }
  }

  /* Text */
  &__text-wrapper {
    display: inline-block;
    transform: skew($skew-angle);
  }

  &__text {
    display: inline-block;
  }

  /* Tagline */
  &__tagline {
    width: 100%;
    margin: t-vertical-rhythm(2) t-vertical-rhythm(1) t-vertical-rhythm(1);
    text-align: center;
    font-family: $s-font-family-quaternary;
    font-size: t-modular-scale(0);
    letter-spacing: 0.4em;
    text-transform: uppercase;
    font-weight: 300;
    color: $s-color-secondary;
    animation: tagline-animation 0.4s 1s ease-out both;

    @keyframes tagline-animation {
      0% {
        transform: translateY(100%);
        opacity: 0;
      }

      50% {
        opacity: 1;
      }

      100% {
        transform: translateY(0);
      }
    }
  }

  &__tagline-breaker {
    display: block;

    @media (min-width: t-breakpoint(md)) {
      display: none;
    }
  }

  /* Language */
  &__language {
    position: absolute;
    bottom: t-vertical-rhythm(1);
    right: t-vertical-rhythm(1);
    color: $s-color-white;
  }

  &__language-link {
    font-size: t-modular-scale(-1);
    font-style: italic;
    color: inherit;
    text-decoration: none;
  }
}
