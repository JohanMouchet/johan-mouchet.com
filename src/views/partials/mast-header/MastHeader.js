// @flow

import * as React from "react";
import { PROFILE } from "constants/profile";
import parse from "utils/parse";
import "./MastHeader.scss";

type Props = {
  firstname: string,
  lastname: string,
  tagline: string,
};

const MastHeader = ({
  firstName = /*html*/ PROFILE.firstName,
  lastName = /*html*/ PROFILE.lastName,
  tagline = /*html*/ `Never stop exploring, <br className="p-mast-header__tagline-breaker" /> never stop creating`,
}: Props) => (
  <header className="p-mast-header">
    <div className="p-mast-header__inner">
      <h1 className="p-mast-header__heading">
        <span className="p-mast-header__half p-mast-header__half--left">
          <span className="p-mast-header__text-wrapper">
            <span className="p-mast-header__text">{firstName}</span>
          </span>
        </span>

        <span className="p-mast-header__line"></span>

        <span className="p-mast-header__half p-mast-header__half--right">
          <span className="p-mast-header__text-wrapper">
            <span className="p-mast-header__text">{lastName}</span>
          </span>
        </span>
      </h1>

      <h2 className="p-mast-header__tagline">{parse(tagline)}</h2>
    </div>
  </header>
);

export default MastHeader;
