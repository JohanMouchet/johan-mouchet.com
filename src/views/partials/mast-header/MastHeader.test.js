import React from "react";
import { shallow } from "enzyme";
import MastHeader from ".";

it("Renders <MastHeader /> without crashing", () => {
  expect(shallow(<MastHeader />)).toMatchSnapshot();
});
