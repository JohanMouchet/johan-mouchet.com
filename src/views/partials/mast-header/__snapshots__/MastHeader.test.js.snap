// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <MastHeader /> without crashing 1`] = `
<header
  className="p-mast-header"
>
  <div
    className="p-mast-header__inner"
  >
    <h1
      className="p-mast-header__heading"
    >
      <span
        className="p-mast-header__half p-mast-header__half--left"
      >
        <span
          className="p-mast-header__text-wrapper"
        >
          <span
            className="p-mast-header__text"
          >
            Johan
          </span>
        </span>
      </span>
      <span
        className="p-mast-header__line"
      />
      <span
        className="p-mast-header__half p-mast-header__half--right"
      >
        <span
          className="p-mast-header__text-wrapper"
        >
          <span
            className="p-mast-header__text"
          >
            Mouchet
          </span>
        </span>
      </span>
    </h1>
    <h2
      className="p-mast-header__tagline"
    >
      Never stop exploring,
      <br
        className="p-mast-header__tagline-breaker"
        key="1"
      />
       never stop creating
    </h2>
  </div>
</header>
`;
